s4cmb/tod.py

```diff
@@ -616,31 +690,55 @@ def compute_simpolangle(self, ch, parallactic_angle, polangle_err=False):
         ...     parallactic_angle=np.array([np.pi] * tod.nsamples))
         >>> assert angles2 is not None
         """
-        if not polangle_err:
-            ang_pix = (90.0 - self.intrinsic_polangle[ch]) * d2r
+        
+        if polangle_err:
+            state_for_polang = np.random.RandomState(self.seed_pa)#+ch) # no need to add ch because already different for each channel as this is for all detectors.
+            
 
-            ## Demodulation or pair diff use different convention
-            ## for the definition of the angle.
-            if not hasattr(self, 'dm'):
-                pol_ang = parallactic_angle + ang_pix + 2.0 * self.hwpangle
+            if self.coherent_pa:
```

Comment:
the `self.coherent_pa` is not useful as it can be reproduced setting to 0 the std. Remove the if and printout.